src/analytic/SRT2DSPECT/SRT2DSPECTReconstruction.cxx

```diff
@@ -0,0 +1,968 @@
+  density_ptr->fill(0);
+  Sinogram<float> sino = proj_data_ptr->get_empty_sinogram(0, 0);
+ // Viewgram<float> view = proj_data_ptr->get_empty_viewgram(0, 0);
+ Viewgram<float> view = proj_data_ptr->get_viewgram(0, 0);
+  if (do_arc_correction)
+    {
+      // need to do this here to get correct dimensions
+      view = arc_correction.do_arc_correction(view);
+    }
+
+  Viewgram<float> view_atten = atten_data_ptr->get_empty_viewgram(0, 0);
+
+  // Retrieve runtime-dependent sizes
+  const int sp = view.get_num_tangential_poss(); //const int sp = proj_data_ptr->get_num_tangential_poss();
+  const int sth = proj_data_ptr->get_num_views();
+  const int sa = proj_data_ptr->get_num_axial_poss(0);
+
+  //	RelatedViewgrams<float> viewgrams;
+
+  const int sx = image.get_x_size();
+  const int sy = image.get_y_size();
+
+  // c ----------------------------------------------
+  // c The rest of the variables used by the program.
+  // c ----------------------------------------------
+  int i, j, k1, k2;
+  int ith,ia, ip, ix1, ix2;   //extra
+  float aux, a, b, f_node;
+  float x; //extra
+
+  const int image_min_x = image.get_min_x();
+  const int image_min_y = image.get_min_y();
+
+  std::vector<float> th(sth, 0), p(sp, 0), x1(sx, 0), x2(sy, 0);
+  std::vector<std::vector<float>> g(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> ddg(sa, std::vector<float>(sp, 0));
+
+  const int Nt = 8, Nmul = sth / Nt;
+  std::vector<float> lg(sp, 0);
+
+  float dh1[Nt], dh2[Nt], t[Nt];
+
+  std::vector<std::vector<float>> hilb(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> fcpe(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> fspe(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> fc(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> fs(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> ddfc(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> ddfs(sa, std::vector<float>(sp, 0));
+
+  std::vector<std::vector<float>> f(sa, std::vector<float>(sp, 0));
+  std::vector<std::vector<float>> ddf(sa, std::vector<float>(sp, 0));
+
+  float rho, h, fcme_fin, fsme_fin, fc_fin, fs_fin, fcpe_fin, fspe_fin, hc_fin, hs_fin, I, Ft1, Ft2, rho1, rho2, tau, tau1, tau2,
+      rx1, rx2;
+  float gx, w, F;
+
+  std::vector<std::vector<float>> lg1_cache(Nt / 2, std::vector<float>(sp - 1, 0));
+  std::vector<std::vector<float>> lg2_cache(Nt / 2, std::vector<float>(sp - 1, 0));
+
+  float*** rx1x2th = reinterpret_cast<float***>(malloc(sa * sizeof(float**)));
+  for (int i = 0; i < sa; i++)
+    {
+      rx1x2th[i] = reinterpret_cast<float**>(malloc(sx * sizeof(float*)));
+      for (int j = 0; j < sx; j++)
+        {
+          rx1x2th[i][j] = reinterpret_cast<float*>(malloc(sy * sizeof(float)));
+          for (int k = 0; k < sy; k++)
+            {
+              rx1x2th[i][j][k] = 0.; // Initialize to zero
+            }
+        }
+    }
```

Comment:
It would not accept 3 integers directly, hence I replaced with: 

IndexRange<3> range(Coordinate3D<int>(0, 0, 0), Coordinate3D<int>(sa - 1, sx - 1, sy - 1));
Array<3, float> rx1x2th(range);
rx1x2th.fill(0.0F);